### PR TITLE
refactor: remove manifest input in validate and build

### PR DESCRIPTION
**Pull request by yiqing-zhao**

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 1]
```diff
@@ -1785,94 +1785,15 @@ export class TeamsAppSolution implements Solution {
...
-                  SolutionError.GetRemoteConfigError
-                )
-              );
-            } else {
-              return err(manifest.error);
-            }
-          }
-        }
-        return await appStudioPlugin.validateManifest(pluginCtx, manifestString);
+        const maybeSelectedPlugins = this.getSelectedPlugins(ctx);
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1788]: this line can be removed?

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 2]
```diff
@@ -1785,94 +1785,15 @@ export class TeamsAppSolution implements Solution {
...
-                  "Solution",
-                  SolutionError.GetRemoteConfigError
-                )
-              );
-            } else {
-              return err(manifest.error);
-            }
-          }
-        }
+        const maybeSelectedPlugins = this.getSelectedPlugins(ctx);
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1793]: as above

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts (hunk 1 above)
Comment (yiqing-zhao) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts]: Thanks! The line is removed now!

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts (hunk 2 above)
Comment (yiqing-zhao) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts]: Thanks! I have removed the line.

**Pull request merged**